From: Henry Su <henry.su@amd.com>
Date: Fri, 11 May 2007 05:48:50 +0000 (-0700)
Subject: Add the combined mode for ATI SB700
X-Git-Tag: v2.6.22~696^2~3
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=823777181b4c0200923dcb026efa5b37f55c0ecf

Add the combined mode for ATI SB700

Besides those modes in ATI SB600 SATA controller, ATI SB700 supports one
more mode:the combined mode.

The combined mode is a Legacy IDE mode used for compatibility with some old
OS without AHCI driver, but now it is not necessary for Linux since the
kernel has supported AHCI.

Signed-off-by: Luugi Marsan <luugi.marsan@amd.com>
Cc: Jeff Garzik <jeff@garzik.org>
Cc: Tejun Heo <htejun@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Jeff Garzik <jeff@garzik.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/drivers/pci/quirks.c linux-source-2.6.18/drivers/pci/quirks.c
--- linux-source-2.6.18.orig/drivers/pci/quirks.c	2007-08-03 01:27:43.634528000 -0600
+++ linux-source-2.6.18/drivers/pci/quirks.c	2007-08-03 01:44:58.149571991 -0600
@@ -856,6 +856,7 @@ static void __devinit quirk_sb600_sata(s
 	}
 }
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP600_SATA, quirk_sb600_sata);
+DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP700_SATA, quirk_sb600_sata);
 
 /*
  *	Serverworks CSB5 IDE does not fully support native mode
diff -urpN linux-source-2.6.18.orig/include/linux/pci_ids.h linux-source-2.6.18/include/linux/pci_ids.h
--- linux-source-2.6.18.orig/include/linux/pci_ids.h	2007-05-23 08:49:59.000000000 -0600
+++ linux-source-2.6.18/include/linux/pci_ids.h	2007-08-03 01:44:58.149571991 -0600
@@ -363,6 +363,7 @@
 #define PCI_DEVICE_ID_ATI_IXP600_SATA	0x4380
 #define PCI_DEVICE_ID_ATI_IXP600_SRAID	0x4381
 #define PCI_DEVICE_ID_ATI_IXP600_IDE	0x438c
+#define PCI_DEVICE_ID_ATI_IXP700_SATA	0x4390
 
 #define PCI_VENDOR_ID_VLSI		0x1004
 #define PCI_DEVICE_ID_VLSI_82C592	0x0005
